src/MyEventEmitter.js

```diff
@@ -1,12 +1,111 @@
 'use strict';
 
-export class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+class MyEventEmitter {
+  constructor() {
+    this.events = {};
+  }
+
+  on(eventName, listener) {
+    if (!this.events[eventName]) {
+      this.events[eventName] = [];
+    }
+
+    this.events[eventName].push(listener);
+
+    return this;
+  }
+
+  once(eventName, listener) {
+    const listenerWrapper = () => {
+      this.off(eventName, listenerWrapper);
+      listener();
+    };
+
+    this.on(eventName, listenerWrapper);
+
+    return this;
+  }
+
+  off(eventName, listener) {
+    if (this.events[eventName]) {
+      const stringifiedListener = JSON.stringify(listener);
+      const listeners = this.events[eventName];
+
+      for (let i = 0; i < listeners.length; i++) {
+        if (JSON.stringify(listeners[i]) === stringifiedListener) {
+          listeners.splice(i, 1);
```

Comment:
A good idea to usage `splice`. But we could find an index another way. If `listeners` is an array, arrays have method `indexOf`, which returns index if found element.  In this case, you don't need to convert your function into JSON. You found just the index and that's all. This solution will work faster. Think about it. If you have questions please ask them in fe_chat